I copied the full failure;

```
#53 [runc 1/1] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod     --mount=type=bind,src=hack/dockerfile/install,target=/tmp/install         PREFIX=/build /tmp/install/install.sh runc
#53 sha256:f3af8dea88f5099d7a5b9408f3b11cf3b8bb8413732fa53c0c4446a1ca777529
#53 6.313 + cd /tmp/tmp.6DDJh7PKrT/src/github.com/opencontainers/runc
#53 6.313 + git checkout -q 12644e614e25b05da6fd08a38ffa0cfe1903fdec
#53 6.544 + '[' -z '' ']'
#53 6.544 + target=static
#53 6.544 + make 'BUILDTAGS=seccomp ' static
#53 6.991 CGO_ENABLED=1 go build -trimpath   -tags "seccomp  netgo osusergo" -ldflags "-w -extldflags -static -X main.gitCommit="12644e614e25b05da6fd08a38ffa0cfe1903fdec" -X main.version=1.0.0-rc93 " -o runc .
#53 12.01 # github.com/opencontainers/runc
#53 12.01 /usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
#53 12.01 /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libseccomp.a(libseccomp_la-api.o): in function `seccomp_api_get':
#53 12.01 (.text+0x10): multiple definition of `seccomp_api_get'; /tmp/go-link-061029346/000020.o:/tmp/tmp.QZjEB1jivx/src/github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go:150: first defined here
#53 12.01 /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libseccomp.a(libseccomp_la-api.o): in function `seccomp_api_set':
#53 12.01 (.text+0x80): multiple definition of `seccomp_api_set'; /tmp/go-link-061029346/000020.o:/tmp/tmp.QZjEB1jivx/src/github.com/opencontainers/runc/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go:155: first defined here
#53 12.01 collect2: error: ld returned 1 exit status
#53 12.01
#53 12.01 make: *** [Makefile:42: static] Error 2
#53 ERROR: executor failed running [/bin/sh -c PREFIX=/build /tmp/install/install.sh runc]: exit code: 2
```

I'm wondering if its because of the cache mounts that are used; https://github.com/moby/moby/blob/083dbe9fcdd08d8b773f2dd89cd6dafc96fe8745/Dockerfile#L217-L220


I recall Go builds with CGO not invalidating cache if there's changes in C code (see, e.g. https://github.com/golang/go/issues/25419 and https://github.com/golang/go/issues/19883).

Currently, the cache-mount does not specify a ID, which means the cache can be shared between stages. Perhaps we need to add an ID to prevent runc trying to reuse the cache from other stages that also have a cache for `/root/.cache/go-build` see the [`Dockerfile syntax`](https://github.com/moby/buildkit/blob/v0.8.2/frontend/dockerfile/docs/syntax.md#run---mounttypecache)

--- a/Dockerfile
+++ b/Dockerfile
@@ -119,6 +119,7 @@ FROM cross-${CROSS} as dev-base
 
 FROM dev-base AS runtime-dev-cross-false
 ARG DEBIAN_FRONTEND
+RUN echo 'deb http://deb.debian.org/debian buster-backports main' > /etc/apt/sources.list.d/backports.list
 RUN --mount=type=cache,sharing=locked,id=moby-cross-false-aptlib,target=/var/lib/apt \
     --mount=type=cache,sharing=locked,id=moby-cross-false-aptcache,target=/var/cache/apt \
         apt-get update && apt-get install -y --no-install-recommends \
@@ -127,7 +128,7 @@ RUN --mount=type=cache,sharing=locked,id=moby-cross-false-aptlib,target=/var/lib
             libapparmor-dev \
             libbtrfs-dev \
             libdevmapper-dev \
-            libseccomp-dev \
+            libseccomp-dev/buster-backports \
             libsystemd-dev \
             libudev-dev
 
@@ -137,15 +138,16 @@ ARG DEBIAN_FRONTEND
 # on non-amd64 systems.
 # Additionally, the crossbuild-amd64 is currently only on debian:buster, so
 # other architectures cannnot crossbuild amd64.
+RUN echo 'deb http://deb.debian.org/debian buster-backports main' > /etc/apt/sources.list.d/backports.list
 RUN --mount=type=cache,sharing=locked,id=moby-cross-true-aptlib,target=/var/lib/apt \
     --mount=type=cache,sharing=locked,id=moby-cross-true-aptcache,target=/var/cache/apt \
         apt-get update && apt-get install -y --no-install-recommends \
             libapparmor-dev:arm64 \
             libapparmor-dev:armel \
             libapparmor-dev:armhf \
-            libseccomp-dev:arm64 \
-            libseccomp-dev:armel \
-            libseccomp-dev:armhf
+            libseccomp-dev:arm64/buster-backports \
+            libseccomp-dev:armel/buster-backports \
+            libseccomp-dev:armhf/buster-backports
 
 FROM runtime-dev-cross-${CROSS} AS runtime-dev
 

